contributions/catalog/models/capability_utilities.py

```diff
@@ -45,43 +46,58 @@ def to_capability(d):
     if not d: return {}
     capability_list = []
 
-    # init capability
-    if isinstance(d['capability_name'], str):
-        capability_list.append(init_capability())
-    else:
-        for _ in range(len(d['capability_name'])):
+    # check how many capabilities are in the given json
+    # this should be checked keys that is as surfix of _number
+    num_cap = 0
```

Comment:
Instead of going through each key and see if each item satisfies pattern "capability_name_" , can we make use of capability_count from contribute.html? No idea if we can pass on arguments from an html page, but just a thought. Same goes for talents as well.

Reply:
unfortunately, the answer is no for me... I don't know the other way than this.